runtime/connectors/s3/s3.go

```diff
@@ -115,13 +116,26 @@ func (c connector) ConsumeAsFiles(ctx context.Context, env *connectors.Env, sour
 	return rillblob.FetchFileNames(ctx, bucketObj, fetchConfigs, url.Path, url.Host)
 }
 
-func getAwsSessionConfig(conf *Config) (*session.Session, error) {
+func getAwsSessionConfig(ctx context.Context, conf *Config, bucket string) (*session.Session, error) {
 	if conf.AWSRegion != "" {
 		return session.NewSession(&aws.Config{
 			Region: aws.String(conf.AWSRegion),
 		})
 	}
-	return session.NewSessionWithOptions(session.Options{
+	sess, err := session.NewSessionWithOptions(session.Options{
 		SharedConfigState: session.SharedConfigEnable,
 	})
+	if err != nil {
+		return nil, err
+	}
+
+	reg, err := s3manager.GetBucketRegion(ctx, sess, bucket, "")
+	if err != nil {
+		return nil, err
+	}
+	if reg != "" {
+		sess.Config.Region = aws.String(reg)
+	}
```

Comment:
What does this accomplish? According to the [docs](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/s3manager/#GetBucketRegion), "If the regionHint is empty, and the ConfigProvider does not have a region value, an error will be returned." – won't this always return an error then?

Reply:
It takes the region from global env. It will throw the error that there is no region if global has no region.

Reply:
I'm still not sure I understand why it's needed. Doesn't the call to `NewSessionWithOptions` load the default region from the users' environment? And if no default region is found, it will error anyway, right?

Reply:
I see the linked issue now. Is there not a config key we can use to support global bucket fetch?

Reply:
I couldn't find one for golang. Searched the V1 and V2 libraries source code.

Reply:
Okay, thanks. Then let's merge this